src/utils/base32.js

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright (C) 2011 by Isaac Wolkerstorfer
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+*/
+// This implementation comes from the project
+// https://github.com/agnoster/base32-js, from this revision:
+// https://raw.githubusercontent.com/agnoster/base32-js/40f5ae8eba3d432cdfdd6a53dae38aaff78ee96d/lib/base32.js
+// and has been adapted for usage in the profiler server.
+// Its licence is the MIT licence: https://raw.githubusercontent.com/agnoster/base32-js/40f5ae8eba3d432cdfdd6a53dae38aaff78ee96d/LICENSE
+//
+// @flow
+
+// This would be the place to edit if you want a different
+// Base32 implementation
+// Note: We edited the original alphabet to add back the "s" and remove the "u",
+// following Crockford's advice to avoid accidental profanity. Also "u" and "v"
+// can be confused.
+// Otherwise the characters "o", "i", "l" are absent.
+const alphabet = '0123456789abcdefghjkmnpqrstvwxyz';
```

Comment:
nitpicking time!

Some choices that exist in other alphabets:
* 's' is removed because it can be confused with '5' (especially in handwritten form). This was the case in the original implementation, I added it back so that 'u' can be removed.
* 'u' and 'v' can be confused in handwritten form too -- but because I removed the "u" here we're fine about that
* z-base-32 changes the order so that the "easier" characters come more often. I wasn't convinced, but [here is some explanation](http://philzimmermann.com/docs/human-oriented-base-32-encoding.txt) if you want to know more.
* in some other implementations they prefer the letters over the digits, keeping 'o' over '0' or 'i' over '1'.
* I believe 'i' in lowercase is maybe quite recognizable indeed: we could remove 's' and add 'i' back.

Reply:
I'm not too opinionated on this.